data_structures/src/transaction.rs

```diff
@@ -824,4 +877,62 @@ mod tests {
             MintTransaction::with_external_address(epoch, reward, own_pkh, Some(external_pkh), 100);
         assert_eq!(expected_mint, mint);
     }
+
+    // VT_weight = N*INPUT_SIZE + M*OUTPUT_SIZE*gamma
+    #[test]
+    fn test_vt_size() {
+        let vt_body =
+            VTTransactionBody::new(vec![Input::default()], vec![ValueTransferOutput::default()]);
+        let vt_tx = VTTransaction::new(vt_body, vec![KeyedSignature::default()]);
+        assert_eq!(INPUT_SIZE + OUTPUT_SIZE * GAMMA, vt_tx.weight());
```

Comment:
I sort of mistrust tests that re-implement the logic of the computations. I can't see how one piece of code would not conform with another that has been written by the same person using the same programming language.

Moreover, I think that any breaking change in `ValueTransferTransaction::weight` would go unnoticed — these tests would simply pass. 

Using static numbers here is also more useful IMO because they can be used as test vectors for other implementations.

Reply:
:melon:

Reply:
We can have both, simply add a line checking the actual value.